pkg/ipmi/ocp/ocp.go

```diff
@@ -286,6 +318,52 @@ func GetOemIpmiDimmInfo(si *smbios.Info) ([]DimmInfo, error) {
 	return info, nil
 }
 
+// Read type 9 from SMBIOS tables and look for SlotDesignation which contains string 'Boot_Drive',
+// and get bus and device number from the matched table to read the Device ID and Vendor ID of
+// the boot drive for sending the IPMI OEM command.
+// This requires the BDF number is correctly set in the type 9 table.
+func GetOemIpmiBootDriveInfo(si *smbios.Info) (*BootDriveInfo, error) {
+	t1, err := si.GetSystemInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	t9, err := si.GetSystemSlots()
+	if err != nil {
+		return nil, err
+	}
+
+	const systemPath = "/sys/bus/pci/devices/"
```

Comment:
This file probably needs to be named ocp_linux.go, since there are linux depdendencies like this in it.

Reply:
Done